core/src/main/java/io/grpc/internal/MessageFramer.java

```diff
@@ -342,7 +340,7 @@ private BufferChainOutputStream() {
 
     @Override
     public void write(int b) throws IOException {
-      singleByte[0] = (byte) b;
+      byte[] singleByte = new byte[]{(byte)b};
```

Comment:
A few things:

* The JVM is heavily optimized for short lived objects.  This is well known
* There is _hope_ that the variable could be stack allocated, not a guarantee.  It just means a better best-case scenario.
* This change is a trade off.  We are optimizing global memory usage, at the cost of single byte writes.  As long as CodedOutputStream (or a user defined marshaller, which does _not_ implement Drainable) uses the other overload, any slowdown will be limited.  

That said, I still think this change is okay, for the aforementioned reasons.  I can do a micro benchmark if you remain unconvinced.